src/Serialization/ParseNodeFromStringTrait.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Microsoft\Kiota\Abstractions\Serialization;
+
+
+use DateInterval;
+use Exception;
+
+/**
+ * This trait contains utility functions for parsing strings to different types.
+ */
+trait ParseNodeFromStringTrait
+{
+    /**
+     * @throws Exception
+     */
+    public function getDateIntervalFromString(string $value): DateInterval
+    {
+        $negativeValPosition = strpos($value, '-');
+        $invert = 0;
+        $str = $value;
+        if ($negativeValPosition !== false && $negativeValPosition === 0) {
+            // Invert the interval
+            $invert = 1;
+            $str = substr($value, 1);
+        }
+        $dateInterval = new DateInterval($str);
```

Comment:
Suggestion:
```php
try {
        $dateInterval = new DateInterval($str);
    } catch (Exception $e) {
        throw new Exception("Invalid date interval string: $value");
    }
```
Make the thrown exception more graceful.

Reply:
The exception thrown is identical to the one you're suggesting. I don't think catching the exception and then rethrowing it is going to make any difference in this case.